qiskit/transpiler/target.py

```diff
@@ -1311,151 +1314,218 @@ def from_configuration(
-                    # Durations if specified manually should override model objects
-                    if instruction_durations is not None:
-                        try:
-                            duration = instruction_durations.get(gate, edge, unit="s")
-                        except TranspilerError:
-                            duration = None
+                                duration = instruction_durations.get(
+                                    inst=name, qubits=list(qubits), unit="s", parameters=None
+                                )
+                            except TranspilerError:
+                                warnings.warn(
+                                    f"Duration of instruction {name} on qubits {qubits} is "
+                                    "not found in `InstructionDurations` passed, "
+                                    "falling back to `BackendProperties` to fetch duration",
+                                    RuntimeWarning,
+                                )
+                                duration = (
+                                    params["gate_length"][0] if "gate_length" in params else None
+                                )
+                        else:
+                            duration = params["gate_length"][0] if "gate_length" in params else None
+
+                        if prop_name_map[name] is None:
+                            prop_name_map[name] = {}
+
+                        prop_name_map[name][qubits] = InstructionProperties(
+                            error=error, duration=duration
+                        )
 
-                    if error is None and duration is None and calibration is None:
-                        gate_properties[edge] = None
-                    else:
-                        gate_properties[edge] = InstructionProperties(
-                            duration=duration, error=error, calibration=calibration
+                        if isinstance(prop_name_map[name], dict) and any(
+                            v is None for v in prop_name_map[name].values()
+                        ):
+                            # Properties provides gate properties only for subset of qubits
+                            # Associated qubit set might be defined by the gate config here
+                            logger.info(
+                                "Gate properties of instruction %s are not provided for every qubits. "
+                                "This gate is ideal for some qubits and the rest is with finite error. "
+                                "Created backend target may confuse error-aware circuit optimization.",
+                                name,
+                            )
+                except BackendPropertyError:
+                    # This gate doesn't report any property
+                    continue
+
+            # Measure instruction property is stored in qubit property
+            prop_name_map["measure"] = {}
```

Comment:
This seems bit inefficient because `Measure` is a circuit `Instruction` with num_qubits of 1. The qubit-wise instruction properties dict is already prepared in [line 1379](https://github.com/Qiskit/qiskit/pull/11461/files#diff-00276708f86c726ae04a2c670e74177eaad6aad971e017951bb5792501d13bf5R1379), but you let GC discard it and set another dict instance here.

Reply:
Apologies, applied suggestion :)